include/qpid/dispatch/vanflow.h

```diff
@@ -57,6 +57,34 @@ typedef enum vflow_record_type {
     VFLOW_RECORD_BIFLOW_APP    = 0x11,  // Bidirectional Application (L7) flow
 } vflow_record_type_t;
 
+#define VFLOW_RECORD_MAX VFLOW_RECORD_BIFLOW_APP // Update this if you add new record types to vflow_record_type
+
+// Most record-types support critical functionality and must be emitted every time the corresponding
+// event occurs. But a few record-types support observability functionality that can be interrupted,
+// if necessary, to prevent excessive memory growth.
+// This array identifies which is which.
+static const bool observability_records[VFLOW_RECORD_MAX + 1] = {
```

Comment:
"Observability" is not the correct characterization for this.  All of VanFlow is about observability.  A better term would be something like "discretionary", or "nonessential"